community/kernel/src/main/java/org/neo4j/graphdb/traversal/FilterableBranchCollisionPolicy.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphdb.traversal;
+
+import org.neo4j.function.Predicate;
+import org.neo4j.function.Predicates;
+import org.neo4j.graphdb.Path;
+
+
+public interface FilterableBranchCollisionPolicy extends BranchCollisionPolicy
```

Comment:
My thought was to not introduce another interface, since if we deprecate `BranchCollisionPolicy`, which we would like to do if we introduce the filterable version, we eventually end up with the interface being called `FilterableBranchCollisionPolicy` when `BranchCollisionPolicy` gets removed. That is probably not something we want.

I've never heard of a user having a custom `BranchCollisionPolicy` and in those rare cases it might be fine fixing that compilation error. That's just my, perhaps flawed perspective.